src/Getnet/API/BrandCryptToken.php

```diff
@@ -0,0 +1,292 @@
+
+    // Bandeira a qual se destina o token: Visa
+    const TYPE_VISA = "VISA_TAVV";
+
+    public $cryptogram;
+
+    public $token_pan_card;
+
+    public $token_expiration_month;
+
+    public $token_expiration_year;
+
+    public $token_status;
+
+    protected $network_token_id;
+
+    protected $transaction_type;
+
+    protected $cryptogram_type;
+
+    protected $amount;
+
+    protected $customer_id;
+
+    protected $email;
+
+    protected $card_brand;
+
+    /**
+     * BrandCryptToken constructor.
+     *
+     * @param string $network_token_id
+     * @param string $transaction_type
+     * @param string $cryptogram_type
+     * @param float $amount
+     * @param string $customer_id
+     * @param string $email
+     * @param string $card_brand
+     * @param Getnet $credencial
+     */
+    public function __construct(
+        $network_token_id,
+        $transaction_type,
+        $cryptogram_type,
+        $amount,
+        $customer_id,
+        $email,
+        $card_brand,
+        Getnet $credencial = null
+    ) {
+        $this->setNetworkTokenId($network_token_id)
+            ->setTransactionType($transaction_type)
+            ->setCryptogramType($cryptogram_type)
+            ->setAmount($amount)
+            ->setCustomerId($customer_id)
+            ->setEmail($email)
+            ->setCardBrand($card_brand);
+
+        if ($credencial) {
+            $this->generateCryptogram($credencial);
+        }
+    }
+
+    /**
+     *
+     * @return mixed
+     */
+    public function getNetworkTokenId()
+    {
+        return $this->network_token_id;
+    }
+
+    /**
+     *
+     * @param mixed $network_token_id
+     * @return BrandCryptToken
+     */
+    public function setNetworkTokenId($network_token_id)
+    {
+        $this->network_token_id = (string) $network_token_id;
+        return $this;
+    }
+
+    /**
+     *
+     * @return mixed
+     */
+    public function getTransactionType()
+    {
+        return $this->transaction_type;
+    }
+
+    /**
+     *
+     * @param mixed $transaction_type
+     * @return BrandCryptToken
+     */
+    public function setTransactionType($transaction_type)
+    {
+        $this->transaction_type = (string) $transaction_type;
+        return $this;
+    }
+
+    /**
+     *
+     * @return mixed
+     */
+    public function getCryptogramType()
+    {
+        return $this->cryptogram_type;
+    }
+
+    /**
+     *
+     * @param mixed $cryptogram_type
+     * @return BrandCryptToken
+     */
+    public function setCryptogramType($cryptogram_type)
+    {
+        $this->cryptogram_type = (string) $cryptogram_type;
+        return $this;
+    }
+
+    /**
+     *
+     * @return mixed
+     */
+    public function getAmount()
+    {
+        return $this->amount;
+    }
+
+    /**
+     *
+     * @param mixed $amount
+     * @return BrandCryptToken
+     */
+    public function setAmount($amount)
+    {
+        $this->amount = (float) $amount;
```

Comment:
Amount precisa ser convertido para centavos (int)
```suggestion
        $this->amount = (int) (string) ($amount * 100);
```

Reply:
Seria bom criar um teste unitário para verificar essas conversões Ex: https://github.com/edson-nascimento/getnet-php/blob/master/tests/TransactionTest.php